### DISCUSSION TITLE
How to enable environment variables on i3 startup?

### DISCUSSION THREAD

**Discussion by vinhocent** [i3/i3#5822] [Q&A]
I've been trying to run a python script that depends on fontawesome (using it for i3bar) but  whenever I had the `exec_always script.py > somelog.txt` 

The log keeps saying that fontawesome is not found.

**Comment by okraits** [comment]
You can use something like that:

`exec_always bash -c 'FOOBAR=foobar ./test.sh'`

**Comment by orestisfl** [comment]
Make sure you export the environmental variables so that they are visible to i3 on its startup. For example, you can set them up in context/xsession or your shell rc

**Reply by vinhocent** [reply]
Yeah I was thinking something like that - but I think I have my python packages installed in a virtual env - so I'm not really sure how to move on from there

**Reply by okraits** [reply]
You can specify the virtualenv python interpreter when running your script like that:

`exec_always bash -c 'FOOBAR=foobar /my/virtualenv/python /my/python/script'`

You can of course set your environment variables in your shell rc file as well, like orestisfl said.
